app/client/components/identity/EmailAndMarketing/OptOutSection.tsx

```diff
@@ -12,30 +12,49 @@ interface OptOutSectionProps {
   clickHandler: ClickHandler;
 }
 
-const optOutFinder = (
-  consents: ConsentOption[],
-  clickHandler: ClickHandler
-) => (id: string) => {
-  const consent = consents.find(c => c.id === id);
-  return (
-    consent && (
-      <MarketingPreference
-        id={consent.id}
-        description={consent.description}
-        selected={consent.subscribed}
-        onClick={clickHandler}
-      />
-    )
-  );
-};
+/**
+ * NOTE:
+ * Only use this method for an OPT OUT consent, eg. "post_optout"
+ * The description of Opt On consents have changed so for UX/UI purposes they are now opt INs
+ * The backend model remains an opt OUT, so we invert the consented/subscribed value here.
+ */
+const optOutFinderAndInverter =
+  (consents: ConsentOption[], clickHandler: ClickHandler) =>
+  (id: string, altDescription?: string) => {
+    const consent = consents.find((c) => c.id === id);
+    return (
+      consent && (
+        <MarketingToggle
+          id={consent.id}
+          title={consent.name}
+          description={consent.description || altDescription} // Not all consents from IDAPI have a description
```

Comment:
The description field on the [Consent model is an Option](https://github.com/guardian/identity/blob/51c6c1b4db68bb4586e07383166a00202d9f5d8f/identity-model/src/main/scala/com/gu/identity/model/Consent.scala#L43), and the [api change](https://github.com/guardian/identity/pull/2079) is now returning consents without. This doesn't seem to break the TS interface and it's served to the client as an "undefined" value and handled line 71 in the `MarketingToggle` component.

But we could amend it in the ConsentOption interface [here](https://github.com/guardian/manage-frontend/blob/88b00f82ea7ee327497f93e979014412158ce85d/app/client/components/identity/models.ts#L60)

Reply:
imo it's worth having the type/interface match the API response, and that avoids the problem of having to check that it's defined

although by changing that type, there may be other cases in the codebase where it's hasn't been handled in a way where it expects undefined values, so may need some refactoring involved

i say try changing the type and then running a type check, if theres lots of errors in the codebase and it doesn't look easy to refactor then leave the type as it was, and use the way you've handled it here.

it if doesn't look too difficult to refactor after the typecheck then it's worth doing that

Reply:
Refactored in the API response and trickled it down the model 
```
interface ConsentAPIResponse {
  id: string;
  description?: string;
  name: string;
  isOptOut: boolean;
  isChannel: boolean;
  isProduct: boolean;
}

export interface ConsentOption {
  id: string;
  description?: string;
  frequency?: string;
  name: string;
  theme?: string;
  type: ConsentOptionType;
  subscribed: boolean;
  identityName?: string;
  isProduct?: boolean;
  isChannel?: boolean;
}
```
Undefined descriptions and other optional fields are now all handled like this where they are displayed in components: 
```
{description && getDescription(description)}
```